src/ocean/util/serialize/contiguous/Deserializer.d

```diff
@@ -523,44 +523,29 @@ struct Deserializer
 
         size_t pos = 0;
 
-        foreach (i, Field; typeof (S.tupleof))
+        foreach (i, _Field; typeof (S.tupleof))
         {
-            // Mixin to avoid -v2 warning about immutable being a D2 keyword
-            version (D_Version2)
-                mixin(`
-                static assert (!is(Field == immutable),
-                    "Cannot deserialize field " ~  FieldName!(i, S)
-                    ~ " of struct " ~ S.stringof ~ " because it is immutable");
-                `);
+            alias StripQualifier!(_Field) Field;
 
             static if (is (Field == struct))
             {
                 This.e.enforceInputSize!(S)(data.length, pos);
 
                 pos += This.countStructArraySizes!(Field)(data[pos .. $], extra_bytes);
             }
-            else static if (is (Field Element : Element[]))
+            else static if (is (Field _Element : _Element[]))
```

Comment:
Perhaps use something like `QualElement`? With `_Element` and `Element` it is easy to confuse which is unqualified.